web_poet/requests.py

```diff
@@ -0,0 +1,120 @@
+"""
+Quick Start
+-----------
+
+.. code-block:: python
+
+    from web_poet import requests
+
+Developers could then use this alongside the ``async/await`` syntax in Python
+to issue additional requests.
+
+
+Use Cases
+---------
+
+Additional requests in Page Objects are hard to avoid since webpages nowadays
+requires some dynamic interactions. Some examples would be:
+
+    - "clicking" a load more button that loads more data.
+    - "scrolling" to paginate an infinite page.
+    - "hovering" that reveals a tool-tip containing additional page info.
+    - etc
+
+In most cases, these are done via AJAX requests.
+
+.. warning::
+
+    Additional requests should not be confused with "crawling" which aims to
+    visit multiple webpages from a given website. Additional requests are simply
+    a means to interact with the website to access more information from it.
+
+
+Use for other frameworks
+------------------------
+
+Please note that on its own, ``web_poet.request`` doesn't do anything. It doesn't
+know how to implement the request on its own. Thus, for frameworks or projects
+wanting to use additional requests in Page Objects, they need to set the
+implementation of how to download things via:
+
+.. code-block:: python
+
+    web_poet.request_backend_var.set(downloader_implementation)
+"""
+
+import asyncio
+import logging
+from contextvars import ContextVar
+from typing import Optional, List, Dict, ByteString, Any, Union
+
+import attr
+
+from web_poet.page_inputs import ResponseData
+
+logger = logging.getLogger(__name__)
+
+
+mapping = Dict[Union[str, ByteString], Any]
+
+# Frameworks that wants to support additional requests in ``web-poet`` should
+# set the appropriate implementation for requesting data.
+request_backend_var: ContextVar = ContextVar("request_backend")
+
+
+class RequestBackendError(Exception):
+    pass
+
+
+@attr.define
+class Request:
+    """Represents a generic HTTP request."""
+
+    url: str
+    method: str = "GET"
+    headers: Optional[mapping] = None
+    body: Optional[str] = None
```

Comment:
I think it should be bytes, not str

Reply:
This could be split into `raw_data: bytes` and `html: str` into a separate `HttpResponseBody` as per our recent discussions.

I've opened up a separate PR in https://github.com/scrapinghub/web-poet/pull/24 to open up the deep discussion on this.

Reply:
We addressed it for Response, not we need to fix it for the Request in some way (which could actually be different).

Reply:
Addressed this in https://github.com/scrapinghub/web-poet/pull/22/commits/a103181060048bffa49a19108c1d881c4d9131b0.